src/com/facebook/buck/parser/UnexpectedFlavorException.java

```diff
@@ -114,13 +118,25 @@ public static UnexpectedFlavorException createWithSuggestions(
     return builder.build();
   }
 
+  private static ImmutableSet<Flavor> getValidFlavors(Flavored flavored, BuildTarget target) {
+    return target.getFlavors().stream()
+        .filter(flavor -> flavored.hasFlavors(ImmutableSet.of(flavor)))
+        .collect(collectingAndThen(toList(), ImmutableSet::copyOf));
```

Comment:
pleas use `MoreCollectors.toImmutableSet()` directly

Reply:
Awesome! Did not know this collector exists =(

Reply:
And again, to make code more consistent I've changed `getInvalidFlavors` to the same approach, since both methods are private and are used in the same context.